src/util/real_algebraic_number_poly_imp.h

```diff
@@ -165,20 +165,27 @@ class RealAlgebraicNumber
   const poly::AlgebraicNumber& getValue() const { return d_value; }
   /** Get the internal value as a non-const reference. */
   poly::AlgebraicNumber& getValue() { return d_value; }
-#else
-  /** Get the internal value as a const reference. */
-  const Rational& getValue() const { return d_value; }
-  /** Get the internal value as a non-const reference. */
-  Rational& getValue() { return d_value; }
-#endif
   /**
-   * Stores the actual real algebraic number.
+   * Convert rational to poly, which returns d_value if it stores the
+   * value of r, otherwise it converts and returns the rational value of r.
    */
+  static poly::AlgebraicNumber convertToPoly(const RealAlgebraicNumber& r);
+#endif
+  /** Get the internal rational value as a const reference. */
+  const Rational& getRationalValue() const { return d_rat; }
+  /** Get the internal rational value as a non-const reference. */
+  Rational& getRationalValue() { return d_rat; }
 #ifdef CVC5_POLY_IMP
+  /**
+   * Whether the value of this real algebraic number is stored in d_value.
+   * Otherwise, it is stored in d_rat.
+   */
+  bool d_isPoly;
```

Comment:
I find the semantics of this flag a bit unfortunate: `isPoly` seems to refer to libpoly, but "poly" is also a somewhat generic mathematical term. Maybe using `isRational` is a bit clearer?